optuna/multi_objective/samplers/_motpe.py

```diff
@@ -0,0 +1,649 @@
+        This splitting strategy consists of the following two steps:
+            1. Nondonation rank-based selection
+            2. Hypervolume subset selection problem (HSSP)-based selection
+
+        Please refer to the `original paper <https://dl.acm.org/doi/abs/10.1145/3377930.3389817>`_
+        for more details.
+        """
+        cvals = np.asarray(config_vals)
+        lvals = np.asarray(loss_vals)
+
+        # Solving HSSP for variables number of times is a waste of time.
+        # We cache the result of splitting.
+        if _SPLITCACHE_KEY in trial.system_attrs:
+            split_cache = trial.system_attrs[_SPLITCACHE_KEY]
+            indices_below = split_cache["indices_below"]
+            weights_below = split_cache["weights_below"]
+            indices_above = split_cache["indices_above"]
+        else:
+            nondomination_ranks = _calculate_nondomination_rank(lvals)
+            n_below = self._gamma(len(lvals))
+            assert 0 <= n_below <= len(lvals)
+
+            indices = np.array(range(len(lvals)))
+            indices_below = np.array([], dtype=int)
+
+            # Nondomination rank-based selection
+            i = 0
+            while len(indices_below) + sum(nondomination_ranks == i) <= n_below:
+                indices_below = np.append(indices_below, indices[nondomination_ranks == i])
+                i += 1
+
+            # Hypervolume subset selection problem (HSSP)-based selection
+            subset_size = n_below - len(indices_below)
+            if subset_size > 0:
+                rank_i_lvals = lvals[nondomination_ranks == i]
+                rank_i_indices = indices[nondomination_ranks == i]
+                worst_point = np.max(rank_i_lvals, axis=0)
+                reference_point = np.maximum(1.1 * worst_point, 0.9 * worst_point)
+                reference_point[reference_point == 0] = EPS
+                selected_indices = self._solve_hssp(
+                    rank_i_lvals, rank_i_indices, subset_size, reference_point
+                )
+                indices_below = np.append(indices_below, selected_indices)
+            assert len(indices_below) == n_below
+
+            indices_above = np.setdiff1d(indices, indices_below)
+
+            attrs = {"indices_below": indices_below, "indices_above": indices_above}
+            if self._weights is _default_weights_above:
+                weights_below = self._calculate_default_weights_below(lvals, indices_below)
+                attrs["weights_below"] = weights_below
+            study._storage.set_trial_system_attr(trial._trial_id, _SPLITCACHE_KEY, attrs)
+
+        below = cvals[indices_below]
+        if self._weights is _default_weights_above:
+            study._storage.set_trial_system_attr(
+                trial._trial_id,
+                _WEIGHTS_BELOW_KEY,
+                lambda _: np.asarray(
+                    [w for w, v in zip(weights_below, below) if v is not None], dtype=float
+                ),
+            )
```

Comment:
I think the `TypeError` will occur here if we use RDB storage because `lambda` is not JSON serializable. Do you think if it is possible to cache `weights_below` as a value of `_WEIGHTS_BELOW_KEY`?